osp/core/utils/schema_validation.py

```diff
@@ -97,9 +98,24 @@ def _load_data_model_from_yaml(data_model_file):
 
 def _check_cuds_object_cardinality(origin_cuds, dest_oclass, rel, constraints):
 
-    actual_cardinality = len(
-        origin_cuds.get(rel=get_entity(rel), oclass=get_entity(dest_oclass))
-    )
+    rel_entity = get_entity(rel)
+
+    if type(rel_entity) == OntologyRelationship:
+        actual_cardinality = len(
+            origin_cuds.get(rel=rel_entity, oclass=get_entity(dest_oclass))
+        )
+    elif type(rel_entity) == OntologyAttribute:
+        # No datatype checking since this is already done when Cuds are
+        # instanciated or imported from a file
+        if rel_entity in origin_cuds.get_attributes():
+            actual_cardinality = 1
+        else:
+            actual_cardinality = 0
```

Comment:
Given that SimPhoNy v3 does not support multi-valued attributes, this is indeed valid.

Below you have alternative suggestions (**not required**).

"One"-liner

```suggestion
        actual_cardinality = (
            1 if rel_entity in origin_cuds.get_attributes() else 0
        )
```

A shorter but more unreadable option: `actual_cardinality = int(rel_entity in origin_cuds.get_attributes())`.

-----

Just for future reference I also leave here what happens in SimPhoNy v3 when the underlying graph contains multi-valued attributes.

```python
In [17]: list(freiburg.session.graph.triples((freiburg.iri, None, None)))  # artificially set-up
Out[17]: 
[(rdflib.term.URIRef('http://www.osp-core.com/cuds#e78978fe-0932-49ab-9223-4ce0c1182918'),
  rdflib.term.URIRef('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'),
  rdflib.term.URIRef('http://www.osp-core.com/city#City')),
 (rdflib.term.URIRef('http://www.osp-core.com/cuds#e78978fe-0932-49ab-9223-4ce0c1182918'),
  rdflib.term.URIRef('http://www.osp-core.com/city#coordinates'),
  rdflib.term.Literal('[0 0]', datatype=rdflib.term.URIRef('http://www.osp-core.com/cuba#_datatypes/VECTOR-INT-2'))),
 (rdflib.term.URIRef('http://www.osp-core.com/cuds#e78978fe-0932-49ab-9223-4ce0c1182918'),
  rdflib.term.URIRef('http://www.osp-core.com/city#name'),
  rdflib.term.Literal('Freiburg')),
 (rdflib.term.URIRef('http://www.osp-core.com/cuds#e78978fe-0932-49ab-9223-4ce0c1182918'),
  rdflib.term.URIRef('http://www.osp-core.com/city#name'),
  rdflib.term.Literal('Strasbourg'))]

In [18]: freiburg.name
Out[18]: 'Freiburg'

In [19]: freiburg.get_attributes()
Out[19]: 
{<OntologyAttribute city.coordinates>: array([0, 0]),
 <OntologyAttribute city.name>: 'Strasbourg'}
```